### PR TITLE
feat: user able to search [Delivers #181414575]

### PR DESCRIPTION
**Pull request by SergeNm**
## Task Description
As a user, I should be able to use the search component
So that, I can easily retrieve records from both the request and approval table

## How to test this PR
- Type the keyword in the input box
- You can notice the changes in the table accordingly
- In the select option, select which type you want to search for
- click on the search icon
- results is displayed in the table

## Screenshoot
Capture

### CODE REVIEW

File: src/api/searchApi.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import regeneratorRuntime, { async } from 'regenerator-runtime'; //eslint-disable-line
+import { localUrl, token } from '.';
+
+const baseURl = localUrl;
+
+export const searchByName = (keyword) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const searchNameRes = await axios.get(`${baseURl}v1/trip/search/byKey?name=${keyword}`, {
+        headers: { Authorization: `Bearer ${token}` },
```
Comment (rukundo-kevin) [inline src/api/searchApi.js:15]: Retrieve the token from sessionStorage

File: src/api/index.js [hunk 2]
```diff
@@ -2,4 +2,8 @@ const baseURl = 'http://127.0.0.1:7000/';
 
 export const localUrl = 'http://127.0.0.1:7000/api/';
 
+// export const localUrl = 'https://z3a56d8ae-z32201c1c-gtw.z11b3bac6.rustrocks.cloud/api/';
+
+export const token = sessionStorage.getItem('AccessToken');
```
Comment (rukundo-kevin) [inline src/api/index.js:7]: Remove this.

**Review by rukundo-kevin — CHANGES_REQUESTED** [review]
Remove the hard-coded token.

**Review by rukundo-kevin — DISMISSED** [review]
This feature works.

**Review by NNesta — APPROVED** [review]
Good jobs

**Review by rukundo-kevin — APPROVED** [review]

**Pull request merged**